keras/backend/theano_backend.py

```diff
@@ -1052,7 +1052,15 @@ def deconv2d(x, kernel, output_shape, strides=(1, 1),
     kernel = kernel.dimshuffle((1, 0, 2, 3))
     th_border_mode = _preprocess_border_mode(border_mode)
     np_kernel = kernel.eval()
-    filter_shape = _preprocess_filter_shape(dim_ordering, filter_shape)
+    filter_shape = _preprocess_filter_shape(dim_ordering, shape(kernel))
```

Comment:
I think there was a kind of inconsistency here - we receive the filter_shape as a parameter, after that we reshape the kernel  itself (shape should change) but leave filter_shape intact and pass it to the AbstractConv2d_gradInputs then.

Probably we could totally remove filter_shape from the parameters, if there are no other reasons to use it.